### PR TITLE
Reimplementation of the `signal_to_noise_ratio()` function.

### PR DESCRIPTION
**Pull request by Feat-FeAR**
S2N should be compliant with the standard definition of _signal_to_noise_ratio_ (see e.g., PMID: 28499413), namely
```math
\textrm{S2N}=\frac{\bar{x}_{1}-\bar{x}_{2}}{s_1+s_2}
```
In particular, notice that:
- _s_ is the **standard deviation** and not the **variance**
- the noise (again the standard deviation _s_) is to be computed _within_ the two groups (_case_ and _control_) separately

### CODE REVIEW

File: gene_ranker/ranking_methods.py [hunk 1]
```diff
@@ -158,27 +158,28 @@ def norm_cohen_d_ranking(dual_dataset: DualDataset) -> pd.DataFrame:
...
+    case = dual_dataset.case.set_index(dual_dataset.on)
+    control = dual_dataset.control.set_index(dual_dataset.on)
+    if case.empty:
+        raise ValueError("Case matrix is empty. Cannot compute fold change.")
+    if control.empty:
+        raise ValueError("Control matrix is empty. Cannot compute fold change.")
+    case_means = case.apply(mean, axis=1)
+    case_SDs = case.apply(std, axis=1)
+    control_means = control.apply(mean, axis=1)
+    control_SDs = control.apply(std, axis=1)
```
Comment (MrHedmad) [inline gene_ranker/ranking_methods.py]: Python follows a `snake_cases` convention for variable names. I'd rename `control_SDs` to `control_stdev`.

File: gene_ranker/ranking_methods.py [hunk 2]
```diff
@@ -158,27 +158,28 @@ def norm_cohen_d_ranking(dual_dataset: DualDataset) -> pd.DataFrame:
...
-    return pd.DataFrame({
-        dual_dataset.on: dual_dataset.merged[dual_dataset.on],
-        "ranking": ratio
-    })
+    # Assume that the values are logged
+    signal = case_means.to_numpy() - control_means.to_numpy()
+    noise = case_SDs.to_numpy() + control_SDs.to_numpy()
+    s2n = signal / noise
 
+    frame = pd.DataFrame({dual_dataset.on: dual_dataset.case[dual_dataset.on], "ranking": s2n})
```
Comment (MrHedmad) [inline gene_ranker/ranking_methods.py]: Delete the `s2n` variable (also, bad name! Don't overshorten variable names. It's not an issue here but if the function was longer, you'd forget what `x` is 30 lines down), and just return the `signal / noise` in the `pd.DataFrame`, perhaps adding indentation for clarity.

**Review by MrHedmad — CHANGES_REQUESTED** [review]
Changes are great and seem correct to me. I've only left two style comments.

After making a change, we should also update the version of GeneRanker in the `__init__.py` file.

File: gene_ranker/ranking_methods.py (hunk 1 above)
Comment (Feat-FeAR) [reply gene_ranker/ranking_methods.py]: perfectly agree

File: gene_ranker/ranking_methods.py (hunk 2 above)
Comment (Feat-FeAR) [reply gene_ranker/ranking_methods.py]: perfectly agree

**Pull request merged**